translation-web/src/main/java/eu/europeana/api/translation/config/TranslationApiAutoconfig.java

```diff
@@ -1,359 +1,380 @@
+
+      ClientOptions clientOptions = ClientOptions.builder().sslOptions(sslOptions).build();
+
+      lettuceClientConfigurationBuilder.clientOptions(clientOptions).useSsl();
+    }
+
+    LettuceClientConfiguration lettuceClientConfiguration =
+        lettuceClientConfigurationBuilder.build();
+
+    RedisConfiguration redisConf = LettuceConnectionFactory
+        .createRedisConfiguration(translationConfig.getRedisConnectionUrl());
+    return new LettuceConnectionFactory(redisConf, lettuceClientConfiguration);
+  }
+
+  private File getTrustoreFile() throws AppConfigurationException {
+
+    String truststorePathConfig = translationConfig.getTruststorePath();
+    if (truststorePathConfig == null) {
+      throw new AppConfigurationException(
+          "A trustore must be provided in configurations when confinguring redis ssl connection");
+    }
+    // allow configurations to use the full path, for backward compatibility
+    final File trustoreFile = getConfigFile(truststorePathConfig);
+    if (!trustoreFile.exists()) {
+      throw new AppConfigurationException(
+          "Invalid config file location: " + trustoreFile.getAbsolutePath());
+    }
+    return trustoreFile;
+  }
+
+  /**
+   * If the input is a full path, the filename will be extracted
+   * 
+   * @param configFile name of the config file
+   * @return the File object for the configFile within the config folder
+   */
+  private File getConfigFile(String configFile) {
+    return new File(translationConfig.getConfigFolder(), FilenameUtils.getName(configFile));
+  }
+
+  @Bean(BeanNames.BEAN_REDIS_TEMPLATE)
+  public RedisTemplate<String, CachedTranslation> getRedisTemplate() throws AppConfigurationException {
+    RedisTemplate<String, CachedTranslation> redisTemplate = new RedisTemplate<>();
+    LettuceConnectionFactory redisConnectionFactory = getRedisConnectionFactory();
+    redisConnectionFactory.afterPropertiesSet();
+    redisTemplate.setConnectionFactory(redisConnectionFactory);
+    redisTemplate.setKeySerializer(new StringRedisSerializer());
+    redisTemplate.setValueSerializer(
+        new Jackson2JsonRedisSerializer<CachedTranslation>(CachedTranslation.class));
+    redisTemplate.afterPropertiesSet();
+    return redisTemplate;
+  }
+
+  @Bean(BeanNames.BEAN_REDIS_CACHE_SERVICE)
+  @ConditionalOnProperty(name = "redis.connection.url")
+  public RedisCacheService getRedisCacheService(@Qualifier(BeanNames.BEAN_REDIS_TEMPLATE) RedisTemplate<String, CachedTranslation> redisTemplate) throws AppConfigurationException {
+    return new RedisCacheService(redisTemplate);
+  }
+
+  @Bean(BeanNames.BEAN_REDIS_MESSAGE_LISTENER_CONTAINER)
+  RedisMessageListenerContainer getRedisMessageListenerContainer() throws AppConfigurationException {
+      RedisMessageListenerContainer container  = new RedisMessageListenerContainer(); 
+      LettuceConnectionFactory redisConnectionFactory = getRedisConnectionFactory();
```

Comment:
this is dupplicated code, we can instntiate the redisConnectionFactory as bean and use it as method parameter with @autowired annotation